Cognite.Simulator.Utils/SimulationRunnerBase.cs

```diff
@@ -252,14 +254,18 @@ public async Task Run(CancellationToken token)
                     }
                     finally
                     {
-                        await StoreRunConfiguration(
-                            calcState,
-                            calcObj,
-                            startTime,
-                            e,
-                            token).ConfigureAwait(false);
-
-                        PublishSimulationRunStatus("IDLE", token);
+                        // the following check was added because the code below was running even for skipped events
+                        if (!skipped)
+                        {
+                            await StoreRunConfiguration(
+                                calcState,
+                                calcObj,
+                                startTime,
+                                e,
+                                token).ConfigureAwait(false);
+                            _logger.LogDebug("Calculation run finished for event {Id}", eventId);
+                            PublishSimulationRunStatus("IDLE", token);
+                        }
```

Comment:
maybe there was a reason we had it run for in finally block, @lprovensi do you remember?

Reply:
This is mainly to save the run configuration even when the simulation run fails with an error. I think Abdullah's check is valid, but it is such an edge case that probably never got caught. Good catch 😉